### PR TITLE
TINY-6649: Accept commas in URL detection

### PR DESCRIPTION
**Pull request by vpyshnenko**
Related Ticket: TINY-6649

Description of Changes:
* Improve link regex in polaris package to accept commas in URL path
* Update polaris version to 4.0.4 since it is used in limbo library as a dependency for preprocessing of pasted text

Pre-checks:
* [x] ~Changelog entry should be added~
* [x] Tests have been added (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

### CODE REVIEW

File: modules/polaris/src/main/ts/ephox/polaris/api/Regexes.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@
   (?:\.[A-Za-z0-9-]+)*
 )
 (?::[0-9]+)?
-(?:\/[-+~=%.()\/\w]*)?
+(?:\/[-+~=%.,()\/\w]*)?
```
Comment (lnewson) [inline modules/polaris/src/main/ts/ephox/polaris/api/Regexes.ts:40]: We should mention about this modification in the section above that lists changes from the original.

- [ ] Update modifications section above

**Review by lnewson — APPROVED** [review]

**Review by lnewson — CHANGES_REQUESTED** [review]
Actually, upon testing this regex I found an issue with including a comma... it'll incorrectly be included in the following (the comma is part of the sentence, not the URL):
> See https://www.tiny.cloud/, for more information

So we need to take that into account and make sure the comma isn't the last character in the regex. Worth noting this also appears to be an issue if the URL is at the end of the sentence, such as:
> See https://www.tiny.cloud.

**Comment by vpyshnenko** [comment]
Attempt to exclude trailing comma at the end of whole  Regexes.link() regexp  causes  Regexes.autolink tests to fail. Turns out autolink function augments link() regexp with the group of trailing symbols. Modification of  just the path group of link() regexp doesn't affect autolink tests and  at least eliminates trailing comma issue in URLs terminating with path component,  thus extending editor's ability from:
- first comma in path breaks URL
to:
- last comma in path breaks URL

**Review by ashfordneil — APPROVED** [review]

File: modules/polaris/src/test/ts/atomic/api/RegexesTest.ts [hunk 2]
```diff
@@ -230,4 +231,36 @@ UnitTest.test('RegexesTest', () => {
       assert.fail('expected ' + v + ' but did not match "' + k + '"');
     }
   });
+  const onlyWithPathLinks = { // Ignore trailing comma and period in URL path
```
Comment (lnewson) [inline modules/polaris/src/test/ts/atomic/api/RegexesTest.ts:234]: Can you please put a newline above this, just to break up the test cases please?

**Review by lnewson — APPROVED** [review]
Thanks for reworking this to prevent the issue where the trailing comma would be included.

**Pull request merged**